ash/src/lib.rs

```diff
@@ -91,6 +91,19 @@ impl<'r, T> RawPtr<T> for Option<&'r T> {
     }
 }
 
+pub trait RawMutPtr<T> {
+    unsafe fn as_raw_mut_ptr(self) -> *mut T;
+}
+
+impl<'r, T> RawMutPtr<T> for Option<&'r mut T> {
+    unsafe fn as_raw_mut_ptr(self) -> *mut T {
```

Comment:
I just want to call out that this doesn't work if you don't consume `self`, because `&mut T` is not `Copy` (that would be aliasing). However, that makes this slightly inconsistent with regards to `RawPtr` (just above).

Reply:
I wonder if `trait RawPtr` should take `self` by value too, given that `Option<&T>` is `Copy` (and we don't have `&Option<&T>`).

Locally this violates `clippy::wrong_self_convention` though.

Reply:
Ah yeah, I missed that lint. I could change both to `to_...` or just do `RawMutPtr::to_raw_mut_ptr` and leave `RawPtr` as it is.

Reply:
Since both are private helpers, I'm not opposed to unifying both into a `to_`.